platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/filesystem/internal/FileSystemManagerFactoryImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.io.filesystem.internal;
+
+import static com.swirlds.common.threading.manager.AdHocThreadManager.getStaticThreadManager;
+
+import com.swirlds.base.time.Time;
+import com.swirlds.common.config.StateCommonConfig;
+import com.swirlds.common.config.singleton.ConfigurationHolder;
+import com.swirlds.common.io.config.FileSystemManagerConfig;
+import com.swirlds.common.io.filesystem.FileSystemManager;
+import com.swirlds.common.io.filesystem.FileSystemManagerFactory;
+import com.swirlds.common.io.utility.RecycleBinImpl;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Path;
+
+public class FileSystemManagerFactoryImpl implements FileSystemManagerFactory {
+
+    private static class InstanceHolder {
+        private static final FileSystemManagerFactoryImpl INSTANCE = new FileSystemManagerFactoryImpl();
+    }
```

Comment:
Why are we providing static access to the file system manager? Static access to things like this is something that I think we should go out of our way to discourage.

If there are currently things that need access to things like the temporary file builder, I suggest that we rename the old static implementation to something like `LegacyTemporaryFileBuilder` and mark it as depreciated. The eventual goal for the static use cases should be to stop accessing things via static stateful variables.

Reply:
As we discussed in Slack:
 * The static access is provided for the Factory, not the FileSystemManager
 * The solution uses both Singleton + Factory patterns as many places in base team code.
 * There is a point in saying that we might not need the factory, and providing the ability to change implementing types provides more flexibility than we would use.
  @hendrikebbers , can you jump in? What do you think about the use of a factory in this case?